src/back/glsl/mod.rs

```diff
@@ -416,7 +416,16 @@ impl<'a, W: Write> Writer<'a, W> {
                 ref members,
             } = ty.inner
             {
-                self.write_struct(handle, members)?
+                // No needed to write a struct that also should be written as a global variable
```

Comment:
could you clarify, why do we write the struct *there* in the first place if we can write it here? Doing everything in this loop seems a bit simpler.

Reply:
The first loop goes through all structs, the second one goes through all global variables. 

For example:
```wgsl
// Type {
//     name: Some(
//         "Light",
//     ),
//     inner: Struct {
//         block: true,
//         members: [
//             StructMember {
//                 name: Some(
//                     "proj",
//                 ),
//                 span: None,
//                 ty: [1],
//             },
//         ],
//     },
// },
[[block]]
struct Light {
    proj: mat4x4<f32>;
};

// Type {
//     name: Some(
//         "Lights",
//     ),
//     inner: Struct {
//         block: true,
//         members: [
//             StructMember {
//                 name: Some(
//                     "data",
//                 ),
//                 span: None,
//                 ty: [3],
//             },
//         ],
//     },
// },
[[block]]
struct Lights {
    data: [[stride(96)]] array<Light>;
};

// GlobalVariable {
//     name: Some(
//         "s_lights",
//     ),
//     class: Storage,
//     binding: Some(
//         Resource {
//             group: 0,
//             binding: 1,
//         },
//     ),
//     ty: [4],
//     init: None,
//     interpolation: None,
//     storage_access: LOAD,
// },
[[group(0), binding(1)]]
var<storage> s_lights: [[access(read)]] Lights;
```

Output GLSL:
```glsl
struct Light {
    mat4x4 proj;
};

readonly buffer Lights_block_0 {
    Light data[];
} _group_0_binding_1;
```

Both structs (`Light` and `Lights`) are `block`, but we don't need to write redundant struct for `Lights`, because it also written as buffer.